packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPVideoPlayer.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if TARGET_OS_OSX
+#import <FlutterMacOS/FlutterMacOS.h>
+#else
+#import <Flutter/Flutter.h>
+#endif
+
+#import <AVFoundation/AVFoundation.h>
+
+#import "FVPDisplayLink.h"
+#import "FVPFrameUpdater.h"
+#import "FVPVideoPlayerPlugin_Test.h"
+
+/// FVPVideoPlayer is responsible for managing video playback using AVPlayer.
+/// It provides methods to control playback, adjust volume, handle seeking, and
+/// notify the Flutter engine about new video frames.
+@interface FVPVideoPlayer ()
+/// The AVPlayerItemVideoOutput associated with this video player.
+@property(readonly, nonatomic, nonnull) AVPlayerItemVideoOutput *videoOutput;
+/// The plugin registrar, to obtain view information from.
+@property(nonatomic, weak, nullable) NSObject<FlutterPluginRegistrar> *registrar;
```

Comment:
This should almost certainly not be public.

Reply:
Right, my bad, I copied the whole interface from `FVPVideoPlayerPlugin.m` file. Since most of the project code was there, I assumed it was meant to be this way. Fixed it now, only 3 properties are left in header file, those are needed